Apps/Common/SIMSolverTS.h

```diff
@@ -128,11 +128,10 @@ template<class T1> class SIMSolverTS : public SIMSolver<T1>
             return 2;
       }
     }
-
-    return 0;
```

Comment:
ok see it now, because the loop never exits or breaks, only returns. Maybe not a good design..